Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -364,36 +373,45 @@ namespace ROS2::Utils
                     outcomeCopyTmp.GetResultCode());
                 if (outcomeCopyTmp)
                 {
-                    // create asset info at destination location using the temporary mesh file
-                    const bool assetInfoOk = CreateSceneManifest(targetPathAssetTmp, targetPathAssetInfo, needsCollider, needsVisual);
+                    const bool needsVisual = (assetReferenceType & ReferencedAssetType::VisualMesh) == ReferencedAssetType::VisualMesh;
+                    const bool needsCollider = (assetReferenceType & ReferencedAssetType::ColliderMesh) == ReferencedAssetType::ColliderMesh;
+                    const bool isMeshFile = (needsVisual || needsCollider);
+
+                    // if the asset is a mesh, create asset info at destination location using the temporary mesh file
+                    const bool assetInfoOk = isMeshFile
+                        ? CreateSceneManifest(targetPathAssetTmp, targetPathAssetInfo, needsCollider, needsVisual)
+                        : true;
```

Comment:
A ternary statement isn't needed here.
A logical-or can be used.
```suggestion
                    // if the asset is a mesh, create asset info at destination location using the temporary mesh file
                    const bool assetInfoOk = !isMeshFile ||
                        CreateSceneManifest(targetPathAssetTmp, targetPathAssetInfo, needsCollider, needsVisual);
```

Reply:
Personally I find the ternary more straightforward to read, I'd prefer to keep it that way.